litecommands-fabric/src/main/java/dev/rollczi/litecommands/fabric/FabricScheduler.java

```diff
@@ -0,0 +1,114 @@
+
+    public FabricScheduler(Supplier<Boolean> isMainThread, int pool) {
+        this.isMainThread = isMainThread;
+        this.mainExecutor = Executors.newSingleThreadScheduledExecutor(runnable -> {
+            Thread thread = new Thread(runnable);
+            thread.setName("scheduler-litecommands-fabric-main");
+            return thread;
+        });
+        AtomicInteger asyncCount = new AtomicInteger();
+        ThreadFactory factory = runnable -> {
+            Thread thread = new Thread(runnable);
+            thread.setName(String.format("scheduler-litecommands-fabric-async-%d", asyncCount.getAndIncrement()));
+
+            return thread;
+        };
+
+        this.asyncExecutor = Executors.newScheduledThreadPool(
+            Math.max(pool, Runtime.getRuntime().availableProcessors()) / 2,
+            factory
+        );
+    }
+
+    @Override
+    public <T> CompletableFuture<T> supplyLater(SchedulerPoll type, Duration delay, ThrowingSupplier<T, Throwable> supplier) {
+        CompletableFuture<T> future = new CompletableFuture<>();
+        SchedulerPoll resolve = type.resolve(SchedulerPoll.MAIN, SchedulerPoll.ASYNCHRONOUS);
+        if (resolve.equals(SchedulerPoll.MAIN) && delay.isZero() && isMainThread.get()) {
+            return tryRun(supplier, future);
+        } else {
+            submit(resolve, delay, supplier, future);
+        }
+        return future;
+    }
+
+    protected static <T> CompletableFuture<T> tryRun(ThrowingSupplier<T, @NotNull Throwable> supplier, CompletableFuture<T> future) {
+        try {
+            future.complete(supplier.get());
+        } catch (Throwable e) {
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    public <T> void submit(SchedulerPoll type, Duration delay, ThrowingSupplier<T, @NotNull Throwable> supplier, CompletableFuture<T> future) {
+        if (type.equals(SchedulerPoll.ASYNCHRONOUS)) {
+            asyncExecutor.schedule(() -> {
+                tryRun(supplier, future);
+            }, delay.toMillis(), TimeUnit.MILLISECONDS);
+        } else if (delay.isZero()) {
+            getExecutor().execute(() -> tryRun(supplier, future));
+        } else {
+            getExecutor().execute(new LaterTask<>(delay, supplier, future));
+        }
+    }
+
+    @Override
+    public void shutdown() {
+        SchedulerExecutorPoolImpl.close(asyncExecutor);
+    }
+
+    public abstract ReentrantThreadExecutor<R> getExecutor();
+
+    private class LaterTask<T> implements Runnable {
+        private final ThrowingSupplier<T, @NotNull Throwable> supplier;
+        private final CompletableFuture<T> future;
+
+        private final long end;
+
+        private LaterTask(Duration delay, ThrowingSupplier<T, @NotNull Throwable> supplier, CompletableFuture<T> future) {
+            this.supplier = supplier;
+            this.future = future;
+            end = System.currentTimeMillis() + delay.toMillis();
```

Comment:
If you want to run the task later, please use scheduled executor.

Reply:
@huanmeng-qwq could you please improve this?

Reply:
> @huanmeng-qwq could you please improve this?

https://github.com/BlackBaroness/LiteCommands/pull/2

Reply:
> > @huanmeng-qwq could you please improve this?
> 
> [BlackBaroness#2](https://github.com/BlackBaroness/LiteCommands/pull/2)

oh, sorry, I didn't notice this